diff -upr /home/alex/cache/motif-os-2.3.4/lib/Xm/List.c ./lib/Xm/List.c
--- /home/alex/cache/motif-os-2.3.4/lib/Xm/List.c	2014-05-31 01:28:22.766522288 +0300
+++ ./lib/Xm/List.c	2014-06-03 01:38:58.765445042 +0300
@@ -78,6 +78,7 @@ static char rcsid[] = "$TOG: List.c /mai
 #define FIX_1390	1
 #define FIX_1365	1
 #define FIX_1362	1
+#define FIX_1636    1
 
 #define	BUTTONDOWN	1
 #define	SHIFTDOWN	2
@@ -8514,6 +8515,11 @@ XmListDeleteAllItems(Widget w)
 
   if (lw->list.items && (lw->list.itemCount > 0))
     {
+#ifdef FIX_1636
+      XmListDeselectAllItems(w);
+      SetSelectionParams(lw);
+#endif
+
       Dimension old_max_height = lw->list.MaxItemHeight;
 
       DrawHighlight(lw, lw->list.CurrentKbdItem, FALSE);
