diff -up openmotif-2.3.3/lib/Xm/TextF.c.motifzone_1552 openmotif-2.3.3/lib/Xm/TextF.c
--- openmotif-2.3.3/lib/Xm/TextF.c.motifzone_1552	2009-10-27 17:10:23.000000000 +0100
+++ openmotif-2.3.3/lib/Xm/TextF.c	2012-10-12 18:36:15.664565788 +0200
@@ -97,6 +97,7 @@ static char rcsid[] = "$TOG: TextF.c /ma
 #include <Xm/XmP.h>
 
 #define FIX_1409
+#define FIX_1552
 
 #if (defined(__FreeBSD__) && (__FreeBSD__ < 4)) || \
     (defined(__APPLE__) || defined(__NetBSD__) || defined(__OpenBSD__))
@@ -1750,8 +1751,12 @@ PaintCursor(XmTextFieldWidget tf)
                                  tf->primitive.shadow_thickness)));          
         }
            if (cursor_width > 0 && cursor_height > 0)
-  	  	XCopyArea(XtDisplay(tf), tf->text.ibeam_off, XtWindow(tf), 
-		      tf->text.save_gc, 0, 0, cursor_width, 
+  	  	XCopyArea(XtDisplay(tf), tf->text.ibeam_off, XtWindow(tf),
+#ifdef FIX_1552
+		      tf->text.save_gc, src_x, 0, cursor_width,
+#else
+		      tf->text.save_gc, 0, 0, cursor_width,
+#endif
 		      cursor_height, x, y);
   	}
   }
