--- openmotif-2.3.3.orig/lib/Xm/XmIm.c	2007-10-15 23:51:10.000000000 +0900
+++ openmotif-2.3.3/lib/Xm/XmIm.c	2012-04-12 11:38:47.007928296 +0900
@@ -888,6 +888,25 @@
   *data = NULL;
 }
 
+/*ARGSUSED*/
+static void
+_XmImInvalidateImWidgetPointer(Widget widget,
+                               XtPointer data)
+{
+  XmImShellInfo   im_info;
+
+  if ((im_info = (XmImShellInfo) data) == NULL)
+    return;
+  
+  if (im_info->im_widget == NULL)
+    return;
+
+  XtRemoveCallback(im_info->im_widget, XtNdestroyCallback,
+                   (XtCallbackProc)_XmImInvalidateImWidgetPointer,
+                   (XtPointer)im_info);
+  im_info->im_widget = NULL;
+}
+
 void 
 _XmImChangeManaged(
 		   Widget vw )
@@ -1129,10 +1148,16 @@
    * focus windows.
    */
   if ((im_info->im_widget == NULL) &&
-      (style & (XIMStatusArea | XIMPreeditArea | XIMPreeditPosition)))
+      (style & (XIMStatusArea | XIMPreeditArea | XIMPreeditPosition))) {
     im_info->im_widget = 
       XtVaCreateWidget("xmim_wrapper", coreWidgetClass,
 		       shell, XmNwidth, 10, XmNheight, 10, NULL);
+    if (im_info->im_widget != NULL)
+      XtAddCallback(im_info->im_widget, XtNdestroyCallback,
+                    (XtCallbackProc)_XmImInvalidateImWidgetPointer,
+                    (XtPointer)im_info);
+
+  }
   
   /* Create the XIC info record. */
   xic_info = XtNew(XmImXICRec);
